### PR TITLE
KFP SDK: Add Documentation on K8s Resource Manipulation

### LINKED ISSUE #757: Kubeflow Pipelines Volume examples
The Kubeflow Pipelines DSL was recently extended to include `PipelineVolume` and `PipelineVolumeSnapshot`. These are an abstraction for the user to manipulate PVCs and VolumeSnapshots for data management purposes, also exposing dependencies according to their usage.

See PR https://github.com/kubeflow/pipelines/pull/926 for further details.

We need and example and doc covering the above feature.

Thanks @DanSanche for suggesting this doc!

### LINKED ISSUE #758: Guide to launching persistent Kubernetes resources through Kubeflow Pipelines
Using a pipeline step to launch long-lived Kubernetes resources seems to be the standard way of doing certain tasks, like launching web servers. If this is recommended, it should be explained in the docs. If there's a better way, it should be mentioned (https://github.com/kubeflow/examples/blob/21f76812ec2e0f1e98827b9dede5350080b7e5d3/pipelines/mnist-pipelines/mnist_pipeline.py#L62)

Thanks @DanSanche for suggesting this doc!

### PR DESCRIPTION
**Pull request by elikatsis**
Add documentation for:
  * ResourceOp
  * VolumeOp
  * PipelineVolume
  * VolumeSnapshotOp

I believe this PR closes both #757 and #758.
Of course, if you have a different opinion, feel free to remove any of the following lines.

Closes #757 
Closes #758 

/assign @sarahmaddox

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1122)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-1122--competent-brattain-de2d6d.netlify.com/docs/pipelines/sdk/manipulate-resources/

File: content/docs/pipelines/sdk/manipulate-resources.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  For instance, check these samples of 
+  [ResourceOps](https://github.com/kubeflow/pipelines/tree/master/samples/core/resource_ops), 
+  [VolumeOps](https://github.com/kubeflow/pipelines/tree/master/samples/core/volume_ops)
+  and 
+  [VolumeSnapshotOps](https://github.com/kubeflow/pipelines/tree/master/samples/core/volume_snapshot_ops).
+* Learn more about the 
+  [Kubeflow Pipelines domain-specific language (DSL)](/docs/pipelines/sdk/dsl-overview/),
+  a set of Python libraries that you can use to specify ML pipelines.
+* For quick iteration, 
+  [build components and pipelines](/docs/pipelines/sdk/build-components/).
```
Comment (sarahmaddox) [inline content/docs/pipelines/sdk/manipulate-resources.md]: Heh, just one issue in this PR and it's on the very last line. :) The link is broken - remove the "s" at the end of "components".

Comment (elikatsis) [reply content/docs/pipelines/sdk/manipulate-resources.md]: Oops! Sorry for that, fixed it.

**Comment by elikatsis** [comment]
>At some stage, we should also link to your new doc from the relevant lines in this doc [...]

Added that, what do you think?

>We should also add links from the ops in the SDK ref docs to your new page [...]

Sure! I'll see to it.

**Comment by sarahmaddox** [comment]
Thanks, looks great!

/lgtm
/approve

**Pull request merged**